src/fu-common-version.c

```diff
@@ -261,6 +269,56 @@ fu_common_version_parse (const gchar *version)
 	return fu_common_version_from_uint32 ((guint32) tmp, FU_VERSION_FORMAT_TRIPLET);
 }
 
+/**
+ * fu_common_version_guess_format:
+ * @version: A version number, e.g. "1.2.3"
+ *
+ * Guesses the version format from the version number. This is only a heuristic
+ * and plugins and components should explicitly set the version format whenever
+ * possible.
+ *
```

Comment:
Maybe as part of this all relevant existing plugins should be modified to set version format to guarantee that there aren't surprises later.

Reply:
Which plugins can we be sure of the version number format? I guess I could add it for ColorHug, but I'd be guessing for the others.

One other thing we could do: if the plugin sets a version format and then sets a version that doesn't match the format we'd show some kind of console warning. For instance, I set `quad` and then set `1.2` as my version number.

Reply:
> we'd show some kind of console warning

Actually, we can't do that -- as some of the version formats are visually compatible but not the same, e.g. `quad` v.s. `intel-me`.